src/invidious/helpers/youtube_api.cr

```diff
@@ -89,16 +89,31 @@ def request_youtube_api_search(search_query : String, params : String, region =
   return _youtube_api_post_json("/youtubei/v1/search", data)
 end
 
+# Requests the youtube/v1/navigation/resolve_url endpoint with the required headers
+#
+# The resulting response is a JSON reply with the resolved url embeded within.
+# ```
+# request_youtube_api_resolve_url("https://youtube.com/c/google")                 # => {"endpoint": {"browseEndpoint": {"params": "EgC4AQA%3D", "browseId":"UCK8sQmJBp8GCxrOtXWBpyEA"}, ...}}
+# request_youtube_api_resolve_url("https://youtube.com/c/aaaaaaaaaaaaaaaaaaaaaa") # => InfoException
+# ```
+def request_youtube_api_resolve_url(url)
+  data = {
+    "context" => make_youtube_api_context("US"),
+    "url"     => url,
```

Comment:
In order to avoid problems/typos, I think it would be preferable to hardcode the `https://www.youtube.com` url in here, and only append a path (`/c/user`, `/user/name`, ...)

Reply:
`resolve_url` can resolve sub-domains of YouTube (`m.youtube.com`), I think for future usage it's better to not hardcode anything as this function could be used for other domains. One example is if a specific URL that we want only work under the `m.youtube.com`.

And we don't know all the domains that this endpoint support, so we can't predict the future utilization of this function in Invidious.

Reply:
mmmh, good point!